crates/utils/src/utils/markdown.rs

```diff
@@ -1,3 +1,131 @@
+use markdown_it::MarkdownIt;
+
+pub mod spoiler_rule;
+
 pub fn markdown_to_html(text: &str) -> String {
-  comrak::markdown_to_html(text, &comrak::ComrakOptions::default())
+  let md = &mut MarkdownIt::new();
+  markdown_it::plugins::cmark::add(md);
+  markdown_it::plugins::extra::add(md);
```

Comment:
Adds some functionality which was turned off by default in `cromak`. Full list: https://github.com/rlidwka/markdown-it.rs/blob/master/src/plugins/extra/mod.rs

If this is not desirable, I can remove this line and have a more custom configuration. I haven't got the UI set up, however the original `markdown-it` documentation suggests that this is turned on by default: https://github.com/markdown-it/markdown-it#syntax-extensions

Lemmy UI config for reference: https://github.com/LemmyNet/lemmy-ui/blob/main/src/shared/utils.ts#L794

Reply:
Emojis can be added later in a separate PR.